datasets/us/ofac/press_releases/crawler.py

```diff
@@ -0,0 +1,300 @@
+            sections.append("".join(buffer))
+            buffer = []
+        # Append current element to buffer
+        elem_html = tostring(elem, encoding="unicode", with_tail=True)
+        buffer.append(elem_html)
+    # Flush any remaining content in the buffer
+    if buffer:
+        sections.append("".join(buffer))
+    return sections
+
+
+def get_or_request_review(context, html_part, section_url, url):
+    review = get_review(context, Defendants, section_url, MIN_MODEL_VERSION)
+    if review is None:
+        prompt_result = run_typed_text_prompt(context, PROMPT, html_part, Defendants)
+        review = request_review(
+            context,
+            section_url,
+            html_part,
+            HTML,
+            "Press Release",
+            url,
+            prompt_result,
+            MODEL_VERSION,
+        )
+    return review
+
+
+def check_something_changed(
+    context: Context,
+    review: Review,
+    article_html: str,
+    article_element: HtmlElement,
+) -> bool:
+    """
+    Returns True if the source content has changed.
+    In that case it also reprompts to log whether the extracted data has changed.
+    """
+    if source_changed(review, article_element):
+        prompt_result = run_typed_text_prompt(context, PROMPT, article_html, Defendants)
+        if model_hash(prompt_result) == model_hash(review.orig_extraction_data):
+            context.log.warning(
+                "The source content has changed but the extracted data has not",
+                url=review.source_url,
+                seen_source_value=review.source_value,
+                new_source_value=article_html,
+            )
+        else:
+            # A new extraction result looks different from the known original extraction
+            context.log.warning(
+                "The extracted data has changed",
+                url=review.source_url,
+                orig_extracted_data=review.orig_extraction_data.model_dump(),
+                prompt_result=prompt_result.model_dump(),
+            )
+        return True
+    else:
+        return False
+
+
+def crawl_item(context, item, date, url, article_name):
+    entity = context.make(item.entity_schema)
+    entity.id = context.make_id(item.name, item.address, item.country)
+    entity.add("name", item.name, origin=DEFAULT_MODEL)
+    if item.address != item.country:
+        entity.add("address", item.address, origin=DEFAULT_MODEL)
+    entity.add("country", item.country, origin=DEFAULT_MODEL)
+    entity.add("alias", item.aliases, origin=DEFAULT_MODEL)
+    entity.add("topics", "reg.action")
+
+    # We use the date as a key to make sure notices about separate actions are separate sanction entities
+    sanction = h.make_sanction(context, entity, date)
+    h.apply_date(sanction, "date", date)
+    sanction.set("sourceUrl", url)
+    sanction.add("sourceUrl", item.related_url, origin=DEFAULT_MODEL)
+    sanction.add("summary", item.notes, origin=DEFAULT_MODEL)
```

Comment:
All these entities will have specific sanctions objects from OFAC, let's not make this messy with this